app/utils/auth0_parser.rb

```diff
@@ -0,0 +1,22 @@
+class Auth0Parser
+  def authenticate_request(request)
+    @request = request
+    auth_token
+  rescue JWT::VerificationError, JWT::DecodeError
+    render json: { errors: ['Not Authenticated'] }, status: :unauthorized
+  end
+
+  def http_token
+    if @request.headers['Authorization'].present?
+      @request.headers['Authorization'].split(' ').last
+    end
+  end
```

Comment:
This sounds like should be part of the Authentication class and not the parser
Authentication is where we're dealing with the request and checking that the header exists.
The parser should only deal with parsing the token string and getting the auth0_id

Reply:
Agreed, I just noticed this while looking into some other refactoring. I'm removing the `if` surrounding the `split` since we're already guaranteed that a header is present since the Authentication class passed the request in here.